src/clinical/service-worker-thread/tasks.ts

```diff
@@ -71,27 +78,67 @@ const sortDocs = (sort: string) => (currentRecord: ClinicalInfo, nextRecord: Cli
   return order;
 };
 
-const mapEntityDocuments = (totalDonors: number, schemas: any, query: ClinicalQuery) => ([
-  entityName,
-  results,
-]: [string, ClinicalInfo[]]) => {
-  if (isEmpty(results)) return undefined;
+const mapEntityDocuments = (
+  totalDonors: number,
+  schemas: any,
+  query: ClinicalQuery,
+  completionStats: CompletionRecord[],
+) => (
+  entity: [string, ClinicalInfo[]],
```

Comment:
Why do entity as a tuple instead of a declared type like:
```
type EntityClinicalInfo = {
  entityName: string;
  clinicalInfo: ClinicalInfo[];
}
```

No performance difference but it will be a lot easier to keep track of the meaning of the two variables when they're named.

Reply:
Oh this is from the `Object.entries`, that explains it. I think the typing you've done here is fine, this makes more sense in context. If you add some comments on the method I think thats sufficient.

Reply:
I ended up using an Object type + removing Object.entries which I'd been thinking of doing for a while. Also cleaned up the map function with new types added.